# -*- coding: utf-8 -*-
from odoo import api, models, fields, _
from odoo.exceptions import UserError


import logging
_logger = logging.getLogger(__name__)


class CharacterizationDetail(models.Model):
    _inherit = ['mail.thread', 'mail.activity.mixin']
    _name = 'lmm.characterization_detail'
    _description = "Characterization Details"
    
    name = fields.Char(
        required=True,
        string=_("Internal Id"),
        default="Autogenerated on Save",
    )

    size = fields.Float(
        digits=(16, 9),
        string=_('Size')
    )  

    volts = fields.Float(
        digits=(16, 9),
        string=_('Volts')
    )      

    @api.model
    def create(self, vals):
        seq = self.env['ir.sequence'].next_by_code('lmm.characterization_detail') or _('New')
        vals['name'] = seq
        return super(CharacterizationDetail, self).create(vals)

    def copy(self, default=None):
        default = dict(default or {})

        copied_count = self.search_count(
            [('name', '=like', u"Copy of {}%".format(self.name))])
        if not copied_count:
            new_name = u"Copy of {}".format(self.name)
        else:
            new_name = u"Copy of {} ({})".format(self.name, copied_count)

        default['name'] = new_name
        return super(CharacterizationDetail, self).copy(default)

    _sql_constraints = [
        ('name_unique',
         'UNIQUE(name)',
         "The characterization id must be unique"),
    ]